packages/stack/blockchain/src/index.js

```diff
@@ -22,57 +21,43 @@ class Blockchain {
     embark.registerActionForEvent("pipeline:generateAll:before", this.addArtifactFile.bind(this));
 
     this.blockchainNodes = {};
-    this.events.setCommandHandler("blockchain:node:register", (clientName, startCb) => {
-      this.blockchainNodes[clientName] = startCb;
+    this.events.setCommandHandler("blockchain:node:register", (clientName, { isStartedFn, launchFn, stopFn }) => {
+      this.blockchainNodes[clientName] = { isStartedFn, launchFn, stopFn };
     });
 
-    this.events.setCommandHandler("blockchain:node:start", async (initialBlockchainConfig, cb) => {
-      this.plugins.emitAndRunActionsForEvent("blockchain:config:modify", initialBlockchainConfig, (err, blockchainConfig) => {
+    this.events.setCommandHandler("blockchain:node:start", (blockchainConfig, cb) => {
+      const self = this;
+      const clientName = blockchainConfig.client;
+      function started() {
+        self.startedClient = clientName;
+        self.events.emit("blockchain:started", clientName);
+      }
+      if (clientName === constants.blockchain.vm) {
+        started();
+        return cb();
+      }
+
+      const client = this.blockchainNodes[clientName];
+
+      if (!client) return cb(`Blockchain client '${clientName}' not found, please register this node using 'blockchain:node:register'.`);
+      if (!client.isStartedFn) return cb(`Blockchain client '${clientName}' has no 'isStarted' function registered, please register one using 'blockchain:node:register'.`);
+      if (!client.launchFn) return cb(`Blockchain client '${clientName}' has no launch function registered, please register one using 'blockchain:node:register'.`);
+      if (!client.stopFn) return cb(`Blockchain client '${clientName}' has no stop function registered, please register one using 'blockchain:node:register'.`);
```

Comment:
Assuming that these are error messages, maybe it makes sense to have those checks already done the moment someone tries to register a blockchain client (https://github.com/embark-framework/embark/pull/1896/files#diff-5c83ab0fcefd182e46bfa4a772fc2fb0R25). 

When we move this code to TypeScript and introduce a proper interface for this, we can then get rid off those safe guard checks in the register function altogether, because the code won't even compile.

Reply:
I assume that you mean once we move the checks to the register function, and then move to TypeScript as well, registering a blockchain without the necessary `isStartedFn`, `launchFn`, and `stopFn` won't compile.

I'll move the checks to the register function instead for now.

Reply:
Done.

Reply:
Yes, that's what I meant!